doc/contributing/maintaining-single-executable-application-support.md

```diff
@@ -0,0 +1,110 @@
+# Maintaining Single Executable Applications support
+
+Support for [single executable applications](https://github.com/nodejs/node/blob/master/doc/contributing/technical-priorities.md#single-executable-applications)
+is on of the key technical priorities identified for the success of Node.js.
+
+## High level strategy
+
+From the [next-10 discussions](https://github.com/nodejs/next-10/blob/main/meetings/summit-nov-2021.md#single-executable-applications)
+there are 2 approaches the projects believes are important to support:
+
+* Compile with node into executable (boxnode approach)
+* Bundled onto exising Node.js exe (pkg approach)
+
+### Compile with node into executable
+
+No specific code within the Node.js project is needed to support the
+option of compiling a bundled application along with Node.js into a single
+executable application.
+
+### Bundled onto Existing Node.js
+
+The project does not plan to provide the complete solution but instead the key
+elements which are required in the Node.js executable in order to enabling
+bundling with the shipping Node.js binaries. This includes:
+
+* Looking for a segment within the executable that holds bundled code.
+* Running the bundled code when such a segement is found.
+
+It is left up to external tools/solutions to:
+
+* Bundled code into a single script that can be executed with -e on
+  the command line.
+* Generating a command line with appropriate options, including -e to
+  run the bundle script.
+* adding a segment to the existing Node.js executable which contains
+  the command line and appropriate headers.
+* re-generating or removing signatures on the resulting executable
+* providing a virtual file system, and hooking it in if needed to
+  support native modules or reading file contents.
+
+## Maintaining
+
+### Compile with node into executable
+
+There are not special consideration needed.
```

Comment:
@addaleax is this true or is there something we should add here to help avoid breaking support for single executable applications that use the boxnode approach?

Reply:
Maintaining the boxednode approach requires maintaining a stable embedder API.